internal/tracker/source/tekton/tekton.go

```diff
@@ -348,23 +351,40 @@ func PreparePackage(i *PreparePackageInput) (*hub.Package, error) {
 	var name, version, description, tektonKind string
 	var annotations map[string]string
 	var tasks []map[string]interface{}
+	var steps []v1beta1.Step
+	var containerImages []*hub.ContainerImage
+
 	switch m := i.Manifest.(type) {
 	case *v1beta1.Task:
 		tektonKind = "task"
 		name = m.Name
 		version = m.Labels[versionLabelTKey]
 		description = m.Spec.Description
 		annotations = m.Annotations
+
+		// Prepare container images
+		ts := m.TaskSpec()
+		var defaults []v1beta1.ParamSpec
+		if len(ts.Params) > 0 {
+			defaults = append(defaults, ts.Params...)
+		}
+		mts := taskrun.ApplyParameters(context.Background(), &ts, &v1beta1.TaskRun{}, defaults...)
+		steps = mts.Steps
 	case *v1beta1.Pipeline:
 		tektonKind = "pipeline"
 		name = m.Name
 		version = m.Labels[versionLabelTKey]
 		description = m.Spec.Description
 		annotations = m.Annotations
-		for _, task := range m.Spec.Tasks {
+
+		// Prepare container images
+		ps := m.PipelineSpec()
+		mps := pipelinerun.ApplyParameters(context.Background(), &ps, &v1beta1.PipelineRun{})
+		for _, mts := range mps.Tasks {
+			steps = append(steps, mts.TaskSpec.Steps...)
```

Comment:
This is the most important one. I think we need to check if `Steps` (and/or `TaskSpec`, you'll probably know better) is nil here before using it. This is causing the tracker to crash when processing the official pipelines catalog (*invalid memory address or nil pointer dereference*). It'd be great if you can test locally both the tasks and pipelines catalogs when you are done to make sure everything is all right 🙂

Reply:
Good catch! I have added nil check for TaskSpec and ran a couple of tests locally. PTAL (and sorry for the bug, new to Go 😢 ).

Reply:
Oh no worries, thats's fine 🙂